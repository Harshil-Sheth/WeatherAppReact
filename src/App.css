  .bg {
    /* The image used */
    background-image: url("bg.jpg");
  
    /* Full height */
    height: 100%;
  
    /* Center and scale the image nicely */
    background-position: center;
    background-repeat: no-repeat;
    background-size: cover;
  }
  .scroll{
    overflow-x: auto!important;
    overflow-y: hidden!important;
  }
  .scroll::-webkit-scrollbar {   
    background-color: rgba(255, 255, 255, 0.45);
    border-radius: 10px;
  }
  
  .scroll::-webkit-scrollbar-track {
    border-radius: 10px;
    /* display: none; */
    background-color: rgba(0, 0, 0, 0.35);
    
  }
  .scroll::-webkit-scrollbar-thumb{
  border-radius: 10px;
    background-color: rgba(0, 0, 0, 0.7);
}

.currentWeatherBox{
  width: 100%;
  display: flex;
  justify-content: space-evenly;
}
.currentWeatherBox > div{
  display: flex;
  /* justify-content: space-around; */
  align-items: center;
}

@media screen and (max-width: 992px) {
  .headingsearch {
    flex-direction: column;
  }
}
@media screen and (max-width: 992px) {
  .header {
    width:100%
  }
}
@media screen and (max-width: 830px) {
  .currentWeather {
    margin-top: 20px;
    width: 720px!important;
  }
}

@media screen and (max-width: 770px) {
  .currentWeather {
    width: 650px!important;
  }
  .CurrentWeatherContainer{
    flex-direction: column;
    height:60px;
    justify-content: space-between;
    margin:10px
  }
}
@media screen and (max-width: 700px) {
  .currentWeather {
    width: 560px!important;
  }
 
}
@media screen and (max-width: 600px) {
  .currentWeather {
    width: 460px!important;
  }
 .weatherHeader{
   margin-left:0!important;
 }
}
@media screen and (max-width: 500px) {
  .currentWeather {
    width: 450px!important;
    padding:20px!important;
  }
 
}
@media screen and (max-width: 600px) {
  .inputsearch {
    width: 400px!important;
  }
}
@media screen and (max-width: 480px) {
  .inputsearch {
    width: 300px!important;
  }
  .currentWeather {
    width: 430px!important;
    /* padding:20px!important; */
  }
}
@media screen and (max-width: 450px) {
  .currentWeather {
    width: 410px!important;
    /* padding:20px!important; */
  }
}
@media screen and (max-width: 400px) {
  .hourlycontainer {
        padding-left: 30px!important;
        padding-right: 30px!important;
  }
}
@media screen and (max-width: 420px) {
  .currentWeather {
    width: 360px!important;
    padding:10px!important;
  }
}

