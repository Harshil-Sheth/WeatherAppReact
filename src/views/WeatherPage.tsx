import { useEffect, useState } from "react";
import CurrentWeather from "../components/CurrentWeather/CurrentWeather";
import Header from "../components/Header/Header";
import Location from "../components/Location/Location";
import Search from "../components/Search/Search";
import DailyWeather from "../components/DailyWeather/DailyWeather";
import HourlyWeather from "../components/HourlyWeather/HourlyWeather";

interface MyLocation {
  Latitude?: number;
  Longitude?: number;
  Area?: string;
  Country?: string;
}

interface Weather {
  Current: Currentweather;
  Daily: Array<Dailyweather>;
  Hourly: Array<Hourlyweather>;
  Minutely?: Array<Object>;
}
interface weather {
  main?: string;
  id?: number;
  icon?: string;
}
interface Currentweather {
  temp?: number;
  feels_like?: number;
  wind_speed?: number;
  visibility?: number;
  humidity?: number;
  weather?: Array<weather>;
}
interface Dailyweather {
  temp?: Object;
  dt: number;
  wind_speed?: number;
  humidity?: number;
  sunrise?: number;
  weather?: Array<weather>;
}
interface Hourlyweather {
  temp?: Object;
  dt: number;
  wind_speed?: number;
  humidity?: number;
  weather?: Array<weather>;
}

const WeatherPage = () => {
  const [location, setLocation] = useState<MyLocation>({
    Latitude: undefined,
    Longitude: undefined,
    Area: undefined,
    Country: undefined,
  });
  const [weather, setWeather] = useState<Weather>({
    Current: { temp: undefined, weather: [{}] },
    Daily: [],
    Hourly: [],
    Minutely: [],
  });

  const [city, setCity] = useState("");
  // console.log(location);
  //361ac82e2185de2e626bb5ffa95f8289
  const API_ID: string = "e3a4e39f4c824772f7d35bc0b095f245";
  const LOCATION_API: string = `http://api.openweathermap.org/geo/1.0/reverse?lat=${location.Latitude}&lon=${location.Longitude}&appid=${API_ID}`;
  const WEATHER_API: string = `https://api.openweathermap.org/data/2.5/onecall?lat=${location.Latitude}&lon=${location.Longitude}&units=metric&appid=${API_ID}`;
  const LATLON_API: string = `http://api.openweathermap.org/geo/1.0/direct?q=${city}&appid=${API_ID}`;
  useEffect(() => {
    if ("geolocation" in navigator) {
      if (location.Latitude === undefined || location.Longitude === undefined) {
        console.log("Available");
        navigator.geolocation.getCurrentPosition(function (position) {
          setLocation({
            ...location,
            Latitude: position.coords.latitude,
            Longitude: position.coords.longitude,
          });
        });
      }
    } else {
      console.log("Not Available");
    }
  }, [location]);

  useEffect(() => {
    if (
      location.Latitude !== undefined &&
      location.Longitude !== undefined &&
      location.Area === undefined
    ) {
      var requestOptions: Object = {
        method: "GET",
        redirect: "follow",
      };

      fetch(LOCATION_API, requestOptions)
        .then((response) => response.json())
        .then((result) =>
          setLocation({
            ...location,
            Area: result[0].name,
            Country: result[0].country,
          })
        )
        .catch((error) => console.log("error", error));

      fetch(WEATHER_API, requestOptions)
        .then((response) => response.json())
        .then((result) =>
          setWeather({
            ...weather,
            Current: result.current,
            Daily: result.daily,
            Hourly: result.hourly,
            Minutely: result.minutely,
          })
        )
        .catch((error) => console.log("error", error));
    }
  }, [LOCATION_API, location, WEATHER_API, weather]);

  useEffect(() => {
    var requestOptions: RequestInit = {
      method: "GET",
      redirect: "follow",
    };

    fetch(LATLON_API, requestOptions)
      .then((response) => response.json())
      .then((result) =>
        {
          var latlon= result.map(({ lat, lon }: any): Object => {
          return { lon, lat };
        })
      setLocation({Longitude:latlon[0].lon,Latitude:latlon[0].lat,Area:latlon[0].name,Country:'IN'})
      }
      )
      .catch((error) => console.log("error", error));
  }, [LATLON_API]);

  return (
    <div>
      <div
        style={{
          display: "flex",
          justifyContent: "space-around",
          paddingTop: "50px",
          paddingBottom: "50px",
          alignItems: "center",
        }}
      >
        <Header />
        <Search setcity={setCity} />
      </div>
      <div
        style={{
          display: "flex",
          justifyContent: "space-around",
          alignItems: "center",
          //   marginLeft:'20px'
        }}
      >
        <Location location={location} />
        <CurrentWeather currentWeather={weather.Current} />
      </div>
      <div>
        <DailyWeather dailyWeather={weather.Daily} />
        <HourlyWeather hourlyWeather={weather.Hourly} />
      </div>
    </div>
  );
};

export default WeatherPage;
