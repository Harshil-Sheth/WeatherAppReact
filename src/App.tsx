import WeatherPage from "./views/WeatherPage";

function App():any {
  return (
    <div className='bg'>
      <WeatherPage/>
    </div>
  );
}

export default App;
