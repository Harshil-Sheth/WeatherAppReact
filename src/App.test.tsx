// import { render, screen } from '@testing-library/react';
import App from './App';
import Enzyme, { shallow, ShallowWrapper } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import WeatherPage from './views/WeatherPage';

Enzyme.configure({ adapter: new Adapter() })

let wrapped: ShallowWrapper;

beforeEach((): void => {
  wrapped = shallow(<App />);
});

describe('Test Case For App', () => {

  it('renders the APP component', (): void => {
    expect(wrapped).toMatchSnapshot();
  });

  it('should render weather page div', () => {
    expect(wrapped.find('.bg').length).toEqual(1);
  })
  it('should render weather page', () => {
    expect(wrapped.find(WeatherPage).length).toEqual(1);
  })
})



// test('renders learn react link', () => {
//   render(<App />);
//   const linkElement = screen.getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });
