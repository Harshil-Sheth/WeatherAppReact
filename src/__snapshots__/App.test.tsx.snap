// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Case For App renders the APP component 1`] = `ShallowWrapper {}`;
